File: sentimental-credit/credit.py
while True:
    CardNumber = input("Number: ")
    ReversedCardNumber = CardNumber[::-1]
    NumberLength = len(CardNumber)
    if NumberLength == 13 or NumberLength == 15 or NumberLength == 16:
        break
    else:
        print("INVALID")
        break

Counter, CheckSum = 0, 0
VISALength = [13, 16]
MASTERCARDLength = 16
AMEXLength = 15
VISAStartDigit = "4"
MASTERCARDStartDigit = ["51", "52", "53", "54", "55"]
AMEXStartDigit = ["34", "37"]

for Num in ReversedCardNumber:
    # second-to-last product Sum
    if Counter % 2 != 0:
        Product = str(int(Num) * 2)
        for Ditgit in Product:
            CheckSum += int(Ditgit)
    else:
        CheckSum += int(Num)
    Counter -= 1
if CheckSum % 10 == 0:
    StartDigit = CardNumber[:2]
    if CardNumber[0] == VISAStartDigit and NumberLength in VISALength:
        print("VISA")
    elif StartDigit in MASTERCARDStartDigit and NumberLength == MASTERCARDLength:
        print("MASTERCARD")
    elif StartDigit in AMEXStartDigit and NumberLength == AMEXLength:
        print("AMEX")
    else:
        print("INVALID")
else:
    print("INVALID")

File: credit/credit.c
#include <cs50.h>
#include <stdio.h>
#include <stdlib.h>


int main(void)
{
    int Digits[16];
    int Digit = 0;
    int Length;
        //Length Check
        Digit = 0;
        long CardNumber = get_long("Number: ");
        long Cache = CardNumber;
        while (Cache>0)
        {
        Digits[Digit] = Cache%10;
        Cache = Cache / 10;
        Digit ++;
        }
    int FirstSum = 0 ;
    int SecondSum = 0 ;
    for (int i = 0; i<Digit; i++)
    {
        if (i%2!=0)
        {
            //second-to-last product Sum
            long Product = Digits[i]*2;
            while (Product>0)
            {
                FirstSum = FirstSum + (Product%10);
                Product = Product / 10;
            }
        }
        else
        {
            //product of other numbers
            SecondSum = SecondSum + Digits[i];
        }
    }
    if ((FirstSum + SecondSum) % 10 == 0)
    {
        int StartNum = Digits[Digit-1]*10 + Digits[Digit-2];
        if (Digits[Digit-1] == 4 && (Digit == 13 || Digit == 16))
        {
            printf("VISA\n");
        }
        else if ((StartNum == 51 ||StartNum == 52 ||StartNum == 53 ||StartNum == 54 ||StartNum == 55) && Digit == 16)
        {
            printf("MASTERCARD\n");
        }
        else if ((StartNum == 34 || StartNum == 37) && Digit == 15)
        {
            printf("AMEX\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else
    {
       printf("INVALID\n");
    }
}

